### ISSUE TITLE
[BUG] Error when running example code UndefVarError: libsgtsnepi not defined 

### ISSUE THREAD

**Issue by NebulaComplex** [JuliaGraphs/Graphs.jl#396]
Julia v1.10.4 on JupyterLab
SGtSNEpi v0.3.6

When I run the example code from https://juliagraphs.org/Graphs.jl/dev/first_steps/plotting/
```
using GLMakie, SGtSNEpi, SNAPDatasets
GLMakie.activate!()

g = loadsnap(:as_caida)
y = sgtsnepi(g)
show_embedding(y;
  A = adjacency_matrix(g),        # show edges on embedding
  mrk_size = 1,                   # control node sizes
  lwd_in = 0.01, lwd_out = 0.001, # control edge widths
  edge_alpha = 0.03               # control edge transparency
)
```
I get the error:

UndefVarError: `libsgtsnepi` not defined

Stacktrace:
 [1] _sgtsnepi_c(P::SparseArrays.SparseMatrixCSC{Float64, Int64}, d::Int64, max_iter::Int64, early_exag::Int64, λ::Int64; Y0::Ptr{Nothing}, np::Int64, h::Float64, bb::Float64, eta::Float64, run_exact::Bool, fftw_single::Bool, alpha::Int64, profile::Bool, drop_leaf::Bool, list_grid_size::Vector{Int64}, par_scheme_grid_thres::Int64)
   @ SGtSNEpi ~/.julia/packages/SGtSNEpi/aDIqm/src/sgtsne.jl:260
 [2] sgtsnepi(A::SparseArrays.SparseMatrixCSC{Float64, Int64}; d::Int64, λ::Int64, max_iter::Int64, early_exag::Int64, Y0::Nothing, profile::Bool, np::Int64, version::SGtSNEpi.SGTSNEPI_VERSION, h::Float64, u::Int64, k::Int64, eta::Float64, alpha::Int64, fftw_single::Bool, exact::Bool, drop_leaf::Bool, list_grid_size::Vector{Int64}, bound_box::Float64, flag_unweighted_to_weighted::Bool, par_scheme_grid_thres::Int64)
   @ SGtSNEpi ./array.jl:0
 [3] sgtsnepi(A::SparseArrays.SparseMatrixCSC{Float64, Int64})
   @ SGtSNEpi ~/.julia/packages/SGtSNEpi/aDIqm/src/sgtsne.jl:125
 [4] sgtsnepi(G::Graphs.SimpleGraphs.SimpleGraph{Int64})
   @ SGtSNEpi ~/.julia/packages/SGtSNEpi/aDIqm/src/sgtsne.jl:98
 [5] top-level scope
   @ In[2008]:4

**Comment by gdalle** [comment]
Hi! Thanks for noticing this. I think it might be an issue on the side of [SGtSNEpi.jl](https://github.com/fcdimitr/SGtSNEpi.jl), maybe @fcdimitr can help?

**Comment by fcdimitr** [comment]
Thank you for the ping, @gdalle!

Hi @NebulaComplex,

The current version of `SGtSNEpi.jl` has the following limitation (see also the README at [the GitHub page](https://github.com/fcdimitr/SGtSNEpi.jl/tree/master)):

> :warning: **SGtSNEpi is currently not working on Windows and native M1 Macs**: Either use WSL2 on Windows or use the package via `Rosetta 2` on the M-processor Macs (that means by using the `x86` and not the `arm64` `julia`).

So, if you are on a Windows system, please try to install julia via WSL2; if you are on an M-processor Apple, please try the `Rosetta 2` version.

We are actively working on removing these dependencies to ensure compatibility across all environments where Julia is supported.

Please let me know if these steps help resolve your issue. If so, we can close this thread in the `Graphs.jl` repository and continue any further discussions on the `SGtSNEpi.jl` issues page. Thank you!

**Comment by fcdimitr** [comment]
Hi @NebulaComplex. 

We just released v0.4.0 which should work natively on Windows and M-processor Macs. Could you give it a try and check if your problem is solved? Thanks!
